docs/src/modules/spring/pages/serialization.adoc

```diff
@@ -1 +1,38 @@
 = Handling Serialization
+
+You do not need to handle serialization for the messages of an Event Source Entity or a Value Entity. On the other hand, Kalix exposes endpoints with JSON messages and you need to handle serialization of their inputs and outputs: if you are using Java classes you need to annotate them with the https://github.com/FasterXML/jackson-annotations#usage-general[proper Jackson annotation] but if you are using Java https://openjdk.org/jeps/395[record] no annotation is needed. 
+
+== When consuming Messages from Topics
+
+When consuming events from a topic deserialization works out of the box. By default Kalix serialize Java objects to JSON when publish to a topic, or when emitting events from an entity, whatever they are value changes or snapshots. Therefore it can deserialize them as JSON when consuming them. 
```

Comment:
```suggestion
When consuming events from a topic deserialization works out of the box. By default Kalix serialize Java objects to JSON when publishing to a topic, or when emitting events from an entity, whatever they are value changes or snapshots. Therefore it can deserialize them as JSON when consuming them. 
```

I find this confusing. The title is about consuming message from topics, but we talk about emitting events, value changes and snapshots.